R/db.R

```diff
@@ -203,19 +215,92 @@ setMethod("sqlCreateTable", "DB2/AIX64",
 
 # Microsoft SQL Server ---------------------------------------------------------
 
-# Simple class prototype to avoid messages about unknown classes from setMethod
+#' Simple class prototype to avoid messages about unknown classes from setMethod
+#' @rdname SQLServer
+#' @usage NULL
 setClass("Microsoft SQL Server", where = class_cache)
 
-# For SQL Server, conn@quote will return the quotation mark, however
-# both quotation marks as well as square bracket are used interchangeably for
-# delimited identifiers.  See:
-# https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver16
-# Therefore strip the brackets first, and then call the DBI method that strips
-# the quotation marks.
-# TODO: the generic implementation in DBI should take a quote char as
-# parameter.
+#' SQL Server specific implementation.
+#'
+#' For SQL Server, conn@quote will return the quotation mark, however
+#' both quotation marks as well as square bracket are used interchangeably for
+#' delimited identifiers.  See:
+#' \url{https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver16}
+#' Therefore strip the brackets first, and then call the DBI method that strips
+#' the quotation marks.
+#' TODO: the generic implementation in DBI should take a quote char as
+#' parameter.
+#'
+#' @rdname SQLServer
+#' @docType methods
+#' @inheritParams DBI::dbUnquoteIdentifier
+#' @usage NULL
 setMethod("dbUnquoteIdentifier", c("Microsoft SQL Server", "SQL"),
   function(conn, x, ...) {
     x <- gsub("(\\[)([^\\.]+?)(\\])", "\\2", x)
     callNextMethod( conn, x, ... )
   })
+
+#' SQL Server specific implementation.
+#'
+#' Local temp tables are stored as
+#' \code{ [tempdb].[dbo].[#name]________(padding using underscores)[numeric identifier] }
+#'
+#' True if:
+#' - If catalog_name is supplied it must equal "temdb" or "%" ( wildcard )
+#' - Name must start with "#" followd by a non-"#" character
+#' @rdname SQLServer
+#' @usage NULL
+`isTempTable.Microsoft SQL Server` <- function(conn, name, ...) {
+  args <- list(...)
+  if ( "catalog_name" %in% names(args) ) {
+    catalog_name <- args[["catalog_name"]]
+    if ( !is.null(catalog_name) &&
+        catalog_name != "%" &&
+        length(catalog_name ) > 0 &&
+        catalog_name != "tempdb" ) {
+      return(FALSE)
+    }
+  }
```

Comment:
FWIW in cases like this, I think it's better to create a helper function and rely on R to do the dot matching, i.e. something like:

```R
mssq_is_temp <- function(..., name, catalog_name = NULL) {
  if ( !is.null(catalog_name) &&
        catalog_name != "%" &&
        length(catalog_name ) > 0 &&
        catalog_name != "tempdb" ) {
      return(FALSE)
    }
  ...
}